pkg/runctl/run_manager.go

```diff
@@ -97,60 +98,88 @@ func (c *runManager) prepareRunNamespace(pipelineRun k8s.PipelineRun) error {
 	}
-	if pullSecretName != "" {
-		secretNames = append(secretNames, pullSecretName)
+
+	secretNames := pipelineRun.GetSpec().Secrets
+	stripTektonAnnotationsFunc := secrets.StripAnnotationsFunc("tekton")
+	secretNames, err = c.copySecrets(secretHelper, secretNames, pipelineRun, nil, stripTektonAnnotationsFunc)
+	if err != nil {
+		return errors.Wrap(err, "failed to copy secrets")
 	}
-	err = c.copySecrets(runNamespace, secretNames, pipelineRun)
+
+	imagePullSecrets := pipelineRun.GetSpec().ImagePullSecrets
+	random, err := utils.RandomAlphaNumString(6)
 	if err != nil {
-		return errors.Wrap(err, "Failed to copy secrets.")
+		return err
+	}
+	transformers := []secrets.SecretTransformerType{
+		stripTektonAnnotationsFunc,
+		secrets.StripAnnotationsFunc("jenkins"),
+		secrets.StripLabelsFunc("jenkins"),
+		secrets.AppendNameSuffixFunc(random),
 	}
 
+	imagePullSecrets, err = c.copySecrets(secretHelper, imagePullSecrets, pipelineRun, secrets.DockerOnly, transformers...)
+	if err != nil {
+		return errors.Wrap(err, "failed to copy imagePullSecrets.")
+	}
 	//Create Service Account in Run Namespace
 	accountManager := k8s.NewServiceAccountManager(c.factory, runNamespace)
 
-	serviceAccount, err := accountManager.CreateServiceAccount(serviceAccountName, scmCloneSecretName, pullSecretName)
+	serviceAccount, err := accountManager.CreateServiceAccount(serviceAccountName, pipelineCloneSecretName, imagePullSecrets)
 	if err != nil {
-		return errors.Wrap(err, "Failed to create service account.")
+		return errors.Wrap(err, "failed to create service account.")
 	}
 
 	//Add Role Binding to Service Account
 	_, err = serviceAccount.AddRoleBinding(runClusterRoleName, runNamespace)
 	if err != nil {
-		return errors.Wrap(err, "Failed to create role binding")
+		return errors.Wrap(err, "failed to create role binding")
 	}
 
 	return nil
 }
 
-func (c *runManager) copySecrets(targetNamespace string, secretNames []string, pipelineRun k8s.PipelineRun) error {
-	for _, secretName := range secretNames {
-		targetClient := c.factory.CoreV1().Secrets(targetNamespace)
-		secret, err := c.secretProvider.GetSecret(secretName)
-		if err != nil {
-			pipelineRun.UpdateResult(v1alpha1.ResultErrorContent)
-			pipelineRun.UpdateMessage(err.Error())
-			return err
-		}
-		createSecret(targetClient, targetNamespace, secretName, secret)
+func (c *runManager) copyPipelinePullSecret(pipelineRun k8s.PipelineRun, secretHelper secretUtils.SecretHelper) (string, error) {
+	pipelineCloneSecret := pipelineRun.GetSpec().JenkinsFile.Secret
+	if pipelineCloneSecret == "" {
+		return "", nil
 	}
-	return nil
+	random, err := utils.RandomAlphaNumString(6)
+	if err != nil {
+		return "", err
+	}
+	repoServer, err := pipelineRun.GetRepoServerURL()
+	if err != nil {
+		return "", err
+	}
+	transformers := []secrets.SecretTransformerType{
+		secrets.StripAnnotationsFunc("jenkins"),
+		secrets.StripLabelsFunc("jenkins"),
+		secrets.AppendNameSuffixFunc(random),
```

Comment:
Actually we don't need to append a suffix on our own, if we use `metadata.generateName` and remove `metadata.name`. This has the benefit that uniqueness is guaranteed.